test/acceptance/rule-test.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const Rule = require('../../lib/rules/base');
+const generateRuleTests = require('../helpers/rule-test-harness');
+
+describe('rule public api', function() {
+  describe('mode === fix', function() {
+    generateRuleTests({
+      plugins: [
+        {
+          name: 'fix-test',
+          rules: {
+            'can-fix': class extends Rule {
+              visitor() {
+                return {
+                  ElementNode(node) {
+                    if (node.tag !== 'MySpecialThing') {
+                      return;
+                    }
+
+                    if (this.mode === 'fix') {
+                      node.tag = 'EvenBettererThing';
+                    } else {
+                      this.log({
+                        isFixable: true,
+                        message: 'Do not use MySpecialThing',
+                        line: node.loc && node.loc.start.line,
+                        column: node.loc && node.loc.start.column,
+                        source: this.sourceForNode(node),
+                      });
+                    }
+                  },
+                };
+              }
+            },
+          },
+        },
+      ],
+
+      name: 'can-fix',
+      config: true,
+
+      fix: [
+        {
+          input: '<MySpecialThing/>',
+          output: '<EvenBettererThing/>',
```

Comment:
With a few examples, the test files might become quite long quite fast. Should we use Jest snapshots for this?

We could change the tests file structure from:
```
tests
├── unit
    ├── rules
        ├── quotes-test.js
```

to:
```
tests
├── unit
     ├── rules
          ├──  quotes
                 ├── linter-test.js
                 ├── a-collection-of-hbs-snippets-to-fix.hbs
                 ├── jest-config-file
                 └── __snapshots__
```

Reply:
I kinda like the input/output in the same file (makes understanding the changes made much easier), I'll look into seeing if we can do `.toMatchInlineSnapshot` though.

Reply:
Ya, I have snapshot support ready to go (can't push due to https://www.githubstatus.com/incidents/q07bfjh7jf1t, but will get the PR up soon).

Reply:
Done in https://github.com/ember-template-lint/ember-template-lint/pull/1174